LocalVariableNameCheck incorrectly flags variables named `_` (see JEP456)

https://openjdk.org/jeps/456 allows `_` to be used as a variable name to signify that it is "unused".

checkstyle issues the following warning on `var _`:

```
Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]Lint(Java)
```

Seems like the [regex in `LocalVariableNameCheck`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheck.java#L74) needs to be updated to support this new language feature.

-----------------------------------------
I have read check documentation: https://checkstyle.org/checks/naming/localvariablename.html#LocalVariableName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> javac src/Test.java
PS D:\CS\test> cat src/Test.java
import java.util.PriorityQueue;
import java.util.Queue;

class Test {
    void test() {
        Queue<Integer> q = new PriorityQueue<>();
        q.add(1);
        while (!q.isEmpty()) {
            var _ = q.poll();   // violation
        }
    }
}
PS D:\CS\test> cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
 <module name="TreeWalker">
    <module name="LocalVariableName">
    </module>
 </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.14.2-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:9:17: Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalVariableName]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

I want this check to allow unnamed variables by default.

---

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheck.java
@@ -41,7 +41,7 @@ import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
  * <li>
  * Property {@code format} - Sets the pattern to match valid identifiers.
  * Type is {@code java.util.regex.Pattern}.
- * Default value is {@code "^[a-z][a-zA-Z0-9]*$"}.
+ * Default value is {@code "^([a-z][a-zA-Z0-9]*|_)$"}.
  * </li>
  * </ul>
  * <p>
@@ -71,7 +71,7 @@ public class LocalVariableNameCheck
 
     /** Creates a new {@code LocalVariableNameCheck} instance. */
     public LocalVariableNameCheck() {
-        super("^[a-z][a-zA-Z0-9]*$");
+        super("^([a-z][a-zA-Z0-9]*|_)$");
     }
 
     /**

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/naming/LocalVariableNameCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/naming/LocalVariableNameCheck.xml
@@ -18,7 +18,7 @@
  initialization expressions&lt;/a&gt;
  in FOR loop if one char variable name is prohibited by {@code format} regexp.</description>
             </property>
-            <property default-value="^[a-z][a-zA-Z0-9]*$"
+            <property default-value="^([a-z][a-zA-Z0-9]*|_)$"
                       name="format"
                       type="java.util.regex.Pattern">
                <description>Sets the pattern to match valid identifiers.</description>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheckTest.java
@@ -50,7 +50,7 @@ public class LocalVariableNameCheckTest
     public void testDefault()
             throws Exception {
 
-        final String pattern = "^[a-z][a-zA-Z0-9]*$";
+        final String pattern = "^([a-z][a-zA-Z0-9]*|_)$";
 
         final String[] expected = {
             "122:13: " + getCheckMessage(MSG_INVALID_PATTERN, "ABC", pattern),
@@ -86,6 +86,20 @@ public class LocalVariableNameCheckTest
         };
         verifyWithInlineConfigParser(
                 getPath("InputLocalVariableNameOneCharInitVarName.java"), expected);
+    }
+
+    @Test
+    public void testUnnamedVariables() throws Exception {
+        final String pattern = "^([a-z][a-zA-Z0-9]*|_)$";
+
+        final String[] expected = {
+            "16:13: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+            "17:13: " + getCheckMessage(MSG_INVALID_PATTERN, "_result", pattern),
+            "33:22: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputLocalVariableNameUnnamedVariables.java"), expected);
+
     }
 
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localvariablename/InputLocalVariableNameUnnamedVariables.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localvariablename/InputLocalVariableNameUnnamedVariables.java
@@ -1,0 +1,40 @@
+/*
+LocalVariableName
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.naming.localvariablename;
+
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+public class InputLocalVariableNameUnnamedVariables {
+
+    public void testLocalVariables() {
+        Queue<Integer> q = new PriorityQueue<>();
+        int __ = 0;   // violation
+        int _result = 0; // violation
+        q.add(1);
+        q.add(2);
+        for (Integer element : q) {
+            var _ = q.poll();   // ok, unnamed variable
+            var _ = q.poll();
+        }
+    }
+
+    public void testLocalVariablesInForLoops() {
+        Queue<Integer> q = new PriorityQueue<>();
+
+        for (Integer _ : q) {
+            var x1 = q.poll();
+            var x2 = q.poll();
+        }
+        for (Integer __ : q) {  // violation
+            var x1 = q.poll();
+            var x2 = q.poll();
+        }
+
+        for (int ab = 0, _ = Integer.valueOf(1); ab < 3; ab++) { }
+    }
+}

--- a/src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheckExamplesTest.java
+++ b/src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheckExamplesTest.java
@@ -33,7 +33,7 @@ public class LocalVariableNameCheckExamplesTest extends AbstractExamplesModuleTe
 
     @Test
     public void testExample1() throws Exception {
-        final String pattern = "^[a-z][a-zA-Z0-9]*$";
+        final String pattern = "^([a-z][a-zA-Z0-9]*|_)$";
         final String[] expected = {
             "15:14: " + getCheckMessage(MSG_INVALID_PATTERN, "VAR", pattern),
             "17:14: " + getCheckMessage(MSG_INVALID_PATTERN, "var_1", pattern),

--- a/src/xdocs/checks/naming/localvariablename.xml
+++ b/src/xdocs/checks/naming/localvariablename.xml
@@ -37,7 +37,7 @@
               <td>format</td>
               <td>Sets the pattern to match valid identifiers.</td>
               <td><a href="../../property_types.html#Pattern">Pattern</a></td>
-              <td><code>&quot;^[a-z][a-zA-Z0-9]*$&quot;</code></td>
+              <td><code>&quot;^([a-z][a-zA-Z0-9]*|_)$&quot;</code></td>
               <td>3.0</td>
             </tr>
           </table>

